### PR TITLE
feat: Support file request APIs

### PR DESCRIPTION
**Pull request by congminh1254**
Closes: SDK-1571

### CODE REVIEW

File: boxsdk/object/file_request.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    """Represents the file request."""
+    _item_type = 'file_request'
+
+    @api_call
+    def copy(
+            self,
+            *,
+            description: Optional[str] = None,
+            expire_time: Union[datetime, str] = None,
+            folder: 'Folder' = None,
```
Comment (lukaszsocha2) [inline boxsdk/object/file_request.py:31]: folder is required in copy method. Make it mandatory

File: test/unit/object/test_file_request.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    file_request = test_file_request.update_info(data=data)
+    mock_box_session.put.assert_called_once_with(expected_url, data=json.dumps(data), headers=None, params=None)
+    assert isinstance(file_request, FileRequest)
+    assert file_request['type'] == test_file_request.object_type
+    assert file_request['id'] == test_file_request.object_id
+    assert file_request['title'] == new_title
+    assert file_request['status'] == StatusState.INACTIVE
+
+
+def test_copy(test_file_request, mock_box_session):
```
Comment (lukaszsocha2) [inline test/unit/object/test_file_request.py:47]: please test also expires_at field. Make it parametrised to check string and datetime object.

File: boxsdk/object/file_request.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    ) -> 'FileRequest':
+        # pylint: disable=arguments-differ
+        """Copy an existing file request already present on one folder, and applies it to another folder.
+
+        :param description:
+            A new description for the file request.
+        :param title:
+            A new title for the file request.
+        :param expire_time:
+            A expiration time which after no longer allows the file request to be submitted.
```
Comment (lukaszsocha2) [inline boxsdk/object/file_request.py:46]: rephrase

File: boxsdk/object/file_request.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        :param folder:
+            The folder to which the file request will be saved.
+        :param require_description:
+            A flag indicating whether the file submitted must have a description.
+        :param require_email:
+            A flag indicating whether the file submitted must have sender email.
+        :param status:
+            The status of the file request.
+        :returns:
+            The copy of the file
```
Comment (lukaszsocha2) [inline boxsdk/object/file_request.py:56]: file_request

File: boxsdk/object/file_request.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            data['title'] = title
+        if expire_time is not None:
+            data['expires_at'] = normalize_date_to_rfc3339_format(expire_time)
+        if require_description is not None:
+            data['is_description_required'] = require_description
+        if require_email is not None:
+            data['is_email_required'] = require_email
+        if status is not None:
+            data['status'] = status
+        print(data)
```
Comment (lukaszsocha2) [inline boxsdk/object/file_request.py:75]: remove

File: boxsdk/object/file_request.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            A flag indicating whether the file submitted must have sender email.
+        :param status:
+            The status of the file request.
+        :returns:
+            The copy of the file
+        """
+        # pylint: disable=arguments-differ
+        url = self.get_url('copy')
+        data = {
+            'folder': {'id': folder.object_id, 'type': 'folder'},
```
Comment (lukaszsocha2) [inline boxsdk/object/file_request.py:61]: 'folder' -> folder.type

File: docs/usage/file_request.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Get a File Request's Information
+------------------------
+
+To get a file request object, first call [`client.file_request(file_request_id)`][file_request] to construct the appropriate [`FileRequest`][file_request_class] object, and then calling [`file_request.get(*, fields=None, headers=None, **kwargs)`][get] will return the [`FileRequest`][file_request_class] object populated with data 
+from the API, leaving the original object unmodified.
+
+<!-- sample get_file_requests_id -->
+```python
+file_request = client.file_request(file_request_id='123456').get()
+print('File request {} on folder {}'.format(file_request.id, file_request.folder.name))
```
Comment (lukaszsocha2) [inline docs/usage/file_request.md:25]: use fstrings

File: docs/usage/file_request.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
+File Requests
```
Comment (lukaszsocha2) [inline docs/usage/file_request.md:1]: Put documentation file to separate branch: `documentation_update`

File: docs/usage/file_request.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+[file_request]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.client.html#boxsdk.client.client.Client.file_request
+[file_request_class]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.file_request.FileRequest
+[get]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.base_object.BaseObject.get
+
+
+Copy a File Request's Information
+---------------------------
+
+To copy a file request to use on another directory,  first call [`client.file_request(file_request_id)`][file_request] to construct the appropriate [`FileRequest`][file_request_class] object, and then calling [`file_request.copy(folder, description=None, title=None, expire_time=None, require_description=None, require_email=None, status=None)`][copy] will return the [`FileRequest`][file_request_class] object populated with data new created file request from the API.
```
Comment (lukaszsocha2) [inline docs/usage/file_request.md:36]: remove: `to use on another directory,  `

**Review by lukaszsocha2 — COMMENTED** [review]
Really nice! Some minor comments to apply

**Review by mwwoda — APPROVED** [review]
Looks cool!

**Pull request merged**